client/apollo/react/src/Form/ItemFile/ItemFileCommon.tsx

```diff
@@ -0,0 +1,110 @@
+import validationIcon from "@material-symbols/svg-400/outlined/check_circle-fill.svg";
+import errorIcon from "@material-symbols/svg-400/outlined/error-fill.svg";
+import visibilityIcon from "@material-symbols/svg-400/outlined/visibility-fill.svg";
+
+import {
+  type ComponentProps,
+  type ComponentPropsWithoutRef,
+  type ComponentType,
+  useId,
+} from "react";
+
+import classNames from "classnames";
+import { ItemMessage } from "../ItemMessage/ItemMessageCommon";
+import { Spinner } from "../../Spinner/SpinnerCommon";
+import { ClickIcon } from "../../ClickIcon/ClickIconCommon";
+import { Icon } from "../../Icon/IconCommon";
+
+export const itemFileStates = {
+  success: "success",
+  error: "error",
+  loading: "loading",
+} as const;
+
+type ItemFileState = keyof typeof itemFileStates;
+
+export type ItemFileCommonProps = {
+  state: ItemFileState;
+  title: string;
+  helper: string;
+  subTitle: string;
+  errorMessage?: string;
+  icon?: string;
```

Comment:
renommer et rendre obligatoire:
```suggestion
  deleteIcon: string;
```